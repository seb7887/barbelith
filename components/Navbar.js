import React from 'react';
import styled from 'styled-components';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Lens from '@material-ui/icons/Lens';

const Navbar = ({ router, pageProps: { auth } }) => {
  const { user = {} } = auth || {};
  return (
    <AppBar position={router.pathname === '/' ? 'fixed' : 'static'}>
      <Toolbar>
        <Lens />
        <Typography variant='h4' component='h1'>
          Barbelith
        </Typography>

        {user._id ? (
          // Auth navigation
          <div>
            <Button>Profile</Button>
            <Button variant='outlined'>Signout</Button>
          </div>
        ) : (
            // UnAuth navigation
            <div>
              <Button>Signup</Button>
              <Button>Signin</Button>
            </div>
          )}
      </Toolbar>
    </AppBar>
  );
}

export default Navbar;
