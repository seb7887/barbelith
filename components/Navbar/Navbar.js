import React from 'react';
import { AppBar, Toolbar, Typography, Button, withStyles } from '@material-ui/core';
import Lens from '@material-ui/icons/Lens';

const Navbar = ({ classes, router, pageProps: { auth } }) => {
  const { user = {} } = auth || {};
  return (
    <AppBar
      className={classes.appBar}
      position={router.pathname === '/' ? 'fixed' : 'static'}
      data-testid='navbar'
    >
      <Toolbar>
        <Lens className={classes.icon} />
        <Typography
          variant='h4'
          component='h1'
          className={classes.toolbarTitle}
        >
          Barbelith
        </Typography>

        {user._id ? (
          // Auth navigation
          <div>
            <Button>Profile</Button>
            <Button variant='outlined'>Sign out</Button>
          </div>
        ) : (
            // UnAuth navigation
            <div>
              <Button>Sign up</Button>
              <Button>Sign in</Button>
            </div>
          )}
      </Toolbar>
    </AppBar>
  );
}

const styles = theme => ({
  appBar: {
    zIndex: theme.zIndex.drawer + 1
  },
  toolbarTitle: {
    flex: 1,
    fontWeight: 'bold'
  },
  icon: {
    marginRight: theme.spacing.unit,
  }
});

export default withStyles(styles)(Navbar);
