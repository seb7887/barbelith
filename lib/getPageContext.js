import { SheetsRegistry } from 'jss';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
import createGenerateClassName from '@material-ui/core/styles/createGenerateClassName';

// App color theme
const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
    suppressDeprecationWarnings: true,
    fontFamily: ['"Segoe UI"', 'sans-serif']
  },
  palette: {
    primary: {
      main: '#fff'
    },
    secondary: {
      main: '#000'
    },
    favoriteIcon: {
      main: '#fc766a'
    },
    type: 'light'
  }
});

function createPageContext() {
  return {
    theme,
    // this is needed in order to deduplicate the injection of css in the page
    sheetsManager: new Map(),
    // this is needed in order to inject the critical css
    sheetsRegistry: new SheetsRegistry(),
    // the standard class name generator
    generateClassName: createGenerateClassName()
  };
}

export default function getPageContext() {
  // make sure to create a new context for every server-side request so that data
  // isn't shared between connections (really really BAD)
  if (!process.browser) {
    return createPageContext();
  }

  // reuse context on the client-side
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
