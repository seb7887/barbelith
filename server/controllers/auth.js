const passport = require('passport');
const db = require('../models');

exports.signup = async (req, res) => {
  const { name, email, password } = req.body;
  const newUser = await new db.User({ name, email, password });
  await db.User.register(newUser, password, (err, user) => {
    if (err) {
      return res.status(500).send(err.message);
    }
    return res.status(200).json(user);
  });
};

exports.signin = (req, res, next) => {
  passport.authenticate('local', (err, user, info) => {
    if (err) {
      return res.status(500).json(err.message);
    }
    if (!user) {
      return res.status(400).json(info.message);
    }

    req.logIn(user, err => {
      if (err) {
        return res.status(500).json(err.message);
      }
    });

    return res.json(user);
  })(req, res, next);
};
