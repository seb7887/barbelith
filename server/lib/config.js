require('dotenv').config();

const dev = process.env.NODE_ENV !== 'production';
const port = process.env.PORT || 3000;

const config = {
  port: process.env.PORT || 3000,
  db: dev ? process.env.DEV_DB : process.env.MONGO_URI,
  url: dev ? `http://localhost:${port}` : process.env.PROD_URL,
  secret: process.env.SESSION_SECRET || 'jwtSecret'
};

module.exports = config;
