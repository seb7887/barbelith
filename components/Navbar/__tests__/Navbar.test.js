/* eslint-env jest */
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';

import { render, fireEvent } from 'react-testing-library';
import Router from 'next/router'
import Navbar from '../Navbar';
import { signoutUser as mockSignoutUser } from '../../../lib/auth';

jest.mock('../../../lib/auth', () => {
  return {
    signoutUser: jest.fn(() => Promise.resolve())
  };
});

Router.router = {
  pathname: '/',
  push: () => { },
  prefetch: () => { }
};

afterEach(() => {
  mockSignoutUser.mockClear();
});

describe('<Navbar/>', () => {
  let auth = {};

  it('renders a minimal nav when signed out', () => {
    const { getByTestId, asFragment } = render(
      <Navbar
        context={Router.router}
        pageProps={auth}
        {...Router}
      />
    );
    expect(getByTestId('navbar').textContent).toContain('Sign up');
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a profile button when signed in', () => {
    auth = {
      auth: {
        user: {
          _id: 1
        }
      }
    };

    const { getByTestId } = render(
      <Navbar
        pageProps={auth}
        {...Router}
      />
    );

    expect(getByTestId('navbar').textContent).toContain('Profile');
  });

  it('sign out when signed in', async () => {
    const { getByTestId } = render(
      <Navbar
        pageProps={auth}
        {...Router}
      />
    );

    // submit form
    const signoutButton = getByTestId('signout-button');
    fireEvent.click(signoutButton);

    expect(mockSignoutUser).toHaveBeenCalledTimes(1);
  });
});
