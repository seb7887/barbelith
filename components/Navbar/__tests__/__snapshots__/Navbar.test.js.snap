// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> renders a minimal nav when signed out 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-4 MuiAppBar-positionFixed-5 MuiAppBar-colorPrimary-11 mui-fixed Navbar-appBar-1"
    data-testid="navbar"
  >
    <div
      class="MuiToolbar-root-40 MuiToolbar-regular-42 MuiToolbar-gutters-41"
    >
      <div
        data-testid="active-link"
      >
        <a
          href="/"
          style="text-decoration: none; margin: 0px; padding: 0px; font-weight: bold; color: rgb(0, 0, 0); transition: all 0.3s;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-44 Navbar-icon-3"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2z"
            />
          </svg>
        </a>
      </div>
      <h1
        class="MuiTypography-root-53 MuiTypography-h4-68 Navbar-toolbarTitle-2"
      >
        Barbelith
      </h1>
      <div>
        <button
          class="MuiButtonBase-root-115 MuiButton-root-89 MuiButton-text-91 MuiButton-flat-94"
          data-testid="signup-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-90"
          >
            <div
              data-testid="active-link"
            >
              <a
                href="/signup"
                style="text-decoration: none; margin: 0px; padding: 0px; font-weight: normal; color: rgb(0, 0, 0); transition: all 0.3s;"
              >
                Sign up
              </a>
            </div>
          </span>
          <span
            class="MuiTouchRipple-root-118"
          />
        </button>
        <button
          class="MuiButtonBase-root-115 MuiButton-root-89 MuiButton-text-91 MuiButton-flat-94"
          data-testid="signin-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-90"
          >
            <div
              data-testid="active-link"
            >
              <a
                href="/signin"
                style="text-decoration: none; margin: 0px; padding: 0px; font-weight: normal; color: rgb(0, 0, 0); transition: all 0.3s;"
              >
                Sign in
              </a>
            </div>
          </span>
          <span
            class="MuiTouchRipple-root-118"
          />
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
